### PR TITLE
Add documentation for Dex

### PR DESCRIPTION
**Pull request by iam-veeramalla**
Signed-off-by: iam-veeramalla <abhishek.veeramalla@gmail.com>

**What type of PR is this?**
> /kind documentation

**What does this PR do / why we need it**:
Adds documentation for:
- Dex Overview
- Dex OpenShift Connector
- Dex GitHub Connector
- Option to Disable Dex Installation

**Which issue(s) this PR fixes**:
#213 #142  #375 

**How to test changes / Special notes to the reviewer**:
Install mkdocs
and run `mkdocs serve`

Screenshot 2021-09-01 at 2 13 53 PM
Screenshot 2021-09-01 at 2 14 05 PM
Screenshot 2021-09-01 at 2 14 14 PM

### CODE REVIEW

File: docs/usage/dex.md [hunk 1]
```diff
@@ -0,0 +1,91 @@
+- [Overview](#overview)
+- [Dex OpenShift OAuth Connector](#dex-openshift-oauth-connector)
+    - [Role Mappings](#role-mappings)
+- [Dex GitHub Connector](#dex-github-connector)
+- [Disable DEX](#disable-dex)
+
+## Overview
+
+Dex can be used to delegate authentication to an external identity providers like GitHub, SAML and others. SSO configuration of Argo CD requires updating the Argo CD CR with [Dex connector](https://dexidp.io/docs/connectors/) settings.
```
Comment (jannfis) [inline docs/usage/dex.md:9]: > to an external identity providers

_Nitpick_: This should be either "to external identity providers" or "to an external identity provider"

File: docs/usage/dex.md [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+
+## Dex OpenShift OAuth Connector
+
+The below section describes how to configure Argo CD SSO using OpenShift connector as an example. Dex makes use of the users and groups defined within OpenShift by querying the platform provided OAuth server.
+
+The `OpenShiftOAuth` property can be used to trigger the operator to auto configure the built-in OpenShift OAuth server. The RBAC `Policy` property is used to give the admin role in the Argo CD cluster to users in the OpenShift `cluster-admins` group.
```
Comment (jannfis) [inline docs/usage/dex.md:15]: > The `OpenShiftOAuth` property

_Nitpick_: `OpenShiftOAuth` should be `openShiftOAuth`

File: docs/usage/dex.md [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+  rbac:
+    defaultPolicy: 'role:readonly'
+    policy: |
+      g, cluster-admins, role:admin
+    scopes: '[groups]'
+```
+
+#### Role Mappings
+
+To have a specific user be properly atrributed with the `role:admin` upon SSO through Openshift, the user needs to be in a **group** with the `cluster-admin` role added. If the user only has a direct `ClusterRoleBinding` to the Openshift role for `cluster-admin`, the ArgoCD role will not map.
```
Comment (jannfis) [inline docs/usage/dex.md:36]: > the ArgoCD role will not map

_Nitpick_: Should be "Argo CD"

File: docs/usage/dex.md [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+
+A quick fix will be to create an `cluster-admins` group, add the user to the group and then apply the `cluster-admin` ClusterRole to the group.
```
Comment (jannfis) [inline docs/usage/dex.md:38]: > A quick fix will be to create an `cluster-admins` group

_Nitpick_: should be "... to create a group named `cluster-admins` ..."

File: docs/usage/dex.md [hunk 5]
```diff
@@ -0,0 +1,91 @@
...
+
+## Dex GitHub Connector
+
+The below section describes how to configure Argo CD SSO using GitHub (OAuth2) as an example, but the steps should be similar for other identity providers.
+
+1. Register the application in the identity provider as explained [here](https://argoproj.github.io/argo-cd/operator-manual/user-management/#1-register-the-application-in-the-identity-provider).
+
+2. Update the Argo CD CR.
+
+In the dex.config key, add the github connector to the connectors sub field. See the Dex [GitHub connector documentation](https://github.com/dexidp/website/blob/main/content/docs/connectors/github.md) for explanation of the fields. A minimal config should populate the clientID, clientSecret generated in Step 1.
```
Comment (jannfis) [inline docs/usage/dex.md:54]: > In the dex.config key

Would make sense to put dex.config as code, such as `dex.config`

File: docs/usage/dex.md [hunk 6]
```diff
@@ -0,0 +1,91 @@
...
+You will very likely want to restrict logins to one or more GitHub organization. In the connectors.config.orgs list, add one or more GitHub organizations. Any member of the org will then be able to login to Argo CD to perform management tasks.
```
Comment (jannfis) [inline docs/usage/dex.md:55]: > In the connectors.config.orgs list

Same here, should go as `connectors.config.orgs`

**Review by jannfis — APPROVED** [review]
Awesome docs, @iam-veeramalla !

LGTM with just a few very minor nitpicks. Feel free to merge either with or without addressing them.

**Comment by iam-veeramalla** [comment]
> Awesome docs, @iam-veeramalla !
> 
> LGTM with just a few very minor nitpicks. Feel free to merge either with or without addressing them.
Thanks for the review ... Will address them

**Pull request merged**